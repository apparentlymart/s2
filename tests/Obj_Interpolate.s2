#-*-perl-*-

layerinfo type = core;
layerinfo "majorversion" = 0;

class BracketWrapper
{
    var  string  text;
    function builtin as_string () : string;
}

class BracketWrapper2
{
    var  string  text;
    function builtin toString () : string;
}

class Foo
{
    var string text;
    function as_string() : string;
}

function Foo::as_string() : string {
    return "_${.text}_";
}

function main()
{
    var BracketWrapper bw = new BracketWrapper;
    $bw.text = "as_string";
    print "Wrapped in brackets: $bw\n";

    var BracketWrapper2 bw2 = new BracketWrapper2;
    $bw2.text = "toString";
    print "Wrapped in brackets: $bw2\n";

    var BracketWrapper bnull;
    print "Null BracketWrapper (builtin): {$bnull}\n";

    var Foo fcontent = new Foo;
    $fcontent.text = "underscores";
    print "Defined local as_string: {$fcontent}\n";

    var Foo fnull;
    print "Null local as_string: {$fnull}\n";
}
