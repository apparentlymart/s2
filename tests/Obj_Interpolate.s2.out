Wrapped in brackets: [as_string]
Wrapped in brackets: [toString]
Null BracketWrapper (builtin): {}
Defined local as_string: {_underscores_}
Null local as_string: {}
Empty lines:


end.
